lenisha/scribe
# Refactor App & Add Consent Modal

## What's been done

### ⚙️ Refactor
- Broke NewNote into separate components (DictatedNotes, SOAPNotes, PatientHandout)
- Made a lot of styling/responsiveness improvements
- Broke all dictation into a custom hook (useDictation)
  - This hook is responsible for starting/stopping the transcription, updating the noteText, getting permission, etc.
- Broke out all note content into their own context since they are shared
  - This context keeps track of the transcribed audio text (noteText), SOAP Note generated text (soapText), the generated patient handout text (handoutText), and the start/end time of the transcribed audio
- Moved some files around into different folders (speech and prompt utility files are now in utils folder)
- Updated all variables declared with `var` to be `let` or `const`
- Made improvements to the microphone, such as updating available microphones when inputs change, as well as asking for permission to use microphone

### 🆕 New Additions
- Added a placeholder Logout button on side menu
- Added a consent modal that pops up asking to confirm patient consent before starting transcribing

### 🛠️ Misc
- Updated some of the meta files with a new title aligning to the project (AI Scribe)
- Various linting improvements
- Various prettier changes
- Added some additional configurations to various config files

## Testing Notes
- Make sure that the app still transcribes audio
- Check that the app asks for permission to use microphone
- Check that the app updates available microphones when clicking the dropdown after adding/removing a microphone
- Ensure modal pops up when clicking on "Start Transcription" asking for patient consent
  - Make sure that the "Start Transcription" button is disabled until the checkbox is selected
  - Make sure the transcription starts
- Ensure that the AI Generate SOAP notes and Patient Handout still work

## Screenshots / Videos

Showing the new responsiveness, modal, buttons, etc. - note that I didn't have access to any resources for the AI generation which is why I didn't show them, but I did for speech which is why I could transcribe


https://github.com/user-attachments/assets/4dd40481-dea1-42e2-a30b-32e9baa1098b
